rmgpy/solver/cstr.pyx

```diff
@@ -767,5 +767,5 @@ cdef class ContinuousStirredTankReactor(ReactionSystem):
                         if ir[j, 2] != -1:
                             pd[ir[j, 2], ip[j, 2]] += deriv
 
-        self.jacobian_matrix = pd + cj * np.identity(num_core_species, np.float64)
+        self.jacobian_matrix = pd + cj * np.identity(num_core_species, np.float64) - F * np.identity(num_core_species, np.float64)
```

Comment:
This change isn't mentioned in the commit title. 

Also I'm pretty confident this doesn't fix the jacobian calculation as most jacobian entries should have units (1/s) while I'm reasonably sure you have units of (mol/s) in the new term.

Reply:
> So I did a quick initial overview of this. The first thing I noticed is that there are a lot of hidden changes in your commits that aren't mentioned or explained in the commit messages.
> 
> I unfortunately also think this design as a whole is unnecessary code duplication. The thermodynamics and rates of this reactor are identical to the liquid reactor it seems that it would make much more sense from a design perspective to add inflow and outflows to the liquid reactor instead of adding an entirely new reactor.
> 
> I also think it would be nice for the inlet and outlet to be able to be specified independently this should involve very few changes in the code, but allow users to run semi-batch and draining liquid reactors in addition to CSTR type reactors.

Thanks for the comment! I think modifying the liquid reactor is a good idea. I will make the change later.

> This change isn't mentioned in the commit title.
> 
> Also I'm pretty confident this doesn't fix the jacobian calculation as most jacobian entries should have units (1/s) while I'm reasonably sure you have units of (mol/s) in the new term.

Yeah I realized this mistake and corrected it in a later commit.